### PR TITLE
removing user from mysql DB

### PR DESCRIPTION
**Pull request by vishnureddy-blink**
https://blinkhealth.atlassian.net/browse/INFRAENG-3481

### CODE REVIEW

File: provisioning/offboard/database.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+if [[ -z $matching_users ]]; then
+    echo "No matching users found."
+    exit 1
+fi
+
+echo "Matching users:"
+echo "$matching_users"
+
+read -p "Enter the exact user you want to delete: " user_to_delete
```
[the remark this answers is not in the document]
Comment (satyen-blink) [reply provisioning/offboard/database.sh:24]: After thought.. if you want to search for users.. put it behind a flag so default is to delete.
and flag is there just to do the search.
It removes a lot of human input prompts that don't need to be there.

Less prompts the better.

File: provisioning/offboard/database.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
```
Comment (satyen-blink) [inline provisioning/offboard/database.sh:1]: Please do not call the script `database.sh` re-name it to something more informative. and DO add lots of doc strings to explain what this thing does.

**Review by satyen-blink — CHANGES_REQUESTED** [review]

**Review by satyen-blink — CHANGES_REQUESTED** [review]
Makes the changes that you see fit - after offline discussion.

At least add doc strings for the next person.

File: provisioning/offboard/mysql-user-remove.sh [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# This script is to remove user from Dev & Prod e-commerce backenend DB
```
Comment (satyen-blink) [inline provisioning/offboard/mysql-user-remove.sh:1]: excellent

File: provisioning/offboard/mysql-user-remove.sh [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+read -p "Enter the offboard user first name : " pattern
+
+matching_users=$(MYSQL_PWD="${DB_PASSWORD}" mysql -u $DB_USER -h $DB_HOST -e "SELECT user FROM mysql.user WHERE user REGEXP '${pattern}';" -sN)
+
+if [[ -z $matching_users ]]; then
+    echo "No matching users found."
+    exit 1
+fi
+
+echo "$matching_users"
```
Comment (satyen-blink) [inline provisioning/offboard/mysql-user-remove.sh:28]: echo "Found: $matching_users" 

just be careful as to what the user sees, are there new lines in the output?

Comment (vishnureddy-blink) [reply provisioning/offboard/mysql-user-remove.sh:28]: No Satyen, just minimised the code as much as possible and added doc strings as per your suggestion . I will write Runbook user deletion in detail .

**Pull request merged**